sqlgen/reflect.go

```diff
@@ -51,16 +51,20 @@ const (
 )
 
 // UnbuildStruct extracts SQL values from a struct
-func UnbuildStruct(table *Table, strct interface{}) []interface{} {
+func UnbuildStruct(table *Table, strct interface{}) ([]interface{}, error) {
 	elem := reflect.ValueOf(strct).Elem()
 	values := make([]interface{}, len(table.Columns))
 
 	for i, column := range table.Columns {
 		val := elem.FieldByIndex(column.Index)
-		values[i] = column.Descriptor.Valuer(val)
+		var err error
+		values[i], err = column.Descriptor.Valuer(val).Value()
```

Comment:
my takeaway: escaping a value is much cheaper than escaping a big struct?

Reply:
Both that and `sql` was going to call `Value` anyway before it sent the value to the sql driver, which means that the other value was going to be escaped either way - might as well do it as soon as possible so we don't allocate two different values